sdk/iot/hub/samples/src/paho_iot_hub_twin_example.c

```diff
@@ -281,50 +357,165 @@ static int subscribe()
+        break;
+
+      // Type: A service update to twin desired properties.
+      // Payload: The desired properties JSON.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+        printf("A twin desired properties message was received.\n");
+        if (message->payloadlen)
+        {
+          printf("Received payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+        }
+        printf("Response status was %d.\n", twin_response.status);
+
+        // Device will update property locally and report that property to service.
+        az_span payload_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
+        if (az_failed(update_property(payload_span)))
+        {
+          printf("Failed to update property locally.\n");
+          break;
+        }
+        if (az_failed(report_property()))
+        {
+          printf("Failed to report property to IoT Hub.\n");
+          break;
+        }
+        increment_property_count();
+    }
+  }
+
+  putchar('\n');
+  MQTTClient_freeMessage(&message);
+  MQTTClient_free(topicName);
+
+  return 1;
 }
 
 static int send_get_twin()
 {
   int rc;
-  printf("Requesting twin document\n");
+  printf("Device requesting twin document from service.\n");
 
-  // Get the topic to send a twin GET publish message.
+  // Get the topic to send a twin GET publish message to service.
   if (az_failed(
           rc = az_iot_hub_client_twin_document_get_publish_topic(
               &client, get_twin_topic_request_id, get_twin_topic, sizeof(get_twin_topic), NULL)))
   {
-    printf("Unable to get twin document publish topic, return code %d\n", rc);
+    printf("Unable to get twin document publish topic, return code %d.\n", rc);
     return rc;
   }
 
-  // Publish the get twin message. This will trigger the service to send back the twin document for
-  // this device. The response is handled in the on_received function.
+  // Publish the twin document request. This will trigger the service to send back the twin document
+  // for this device. The response is handled in the on_received function.
   if ((rc = MQTTClient_publish(mqtt_client, get_twin_topic, 0, NULL, 0, 0, NULL))
       != MQTTCLIENT_SUCCESS)
   {
-    printf("Failed to publish twin document request, return code %d\n", rc);
+    printf("Failed to publish twin document request, return code %d.\n", rc);
+    return rc;
+  }
+  return rc;
+}
+
+static int report_property()
+{
+  int rc;
+  az_span reported_property_payload;
+  printf("Device building reported property payload and sending to service.\n");
+
+  if (az_failed(
+          rc = build_reported_property(&reported_property_payload)))
+  {
+    printf("Unable to build reported property payload to send, return code %d.\n", rc);
     return rc;
   }
+  if (az_failed(
+          rc = send_reported_property(reported_property_payload)))
+  {
+    printf("Faied to send reported property payload to service. return code %d.\n", rc);
```

Comment:
nit: consider having a `return rc;` call here too. If code gets added after this function call, unexpected behavior could arise